### DISCUSSION TITLE
英语单词的词形还原（Lemmatization）

### DISCUSSION THREAD

**Discussion by chen3feng** [chen3feng/article#85] [Show and tell]
# 英语单词的词形还原（Lemmatization）

## 定义

把单词的各种时态人称单复数的变形转为原形，比如：

- 动词原型 `calls`, `called`, `calling` -> `call`
- 名词单数 `cats` -> cat
- 比较级和最高级 `hardest` -> `hard`

## 可用的库：

pyhton:

- 自然语言工具包[nltk](https://www.nltk.org/)，[Lemma部分的文档](https://www.datacamp.com/community/tutorials/stemming-lemmatization-python)
- [pattern](https://github.com/clips/pattern) 库
- [MorTransformation](https://github.com/Zhangtd/MorTransformation)，一个很简单的实现

Javascript:

- [JavaScript Lemmatizer](https://www.npmjs.com/package/javascript-lemmatizer)

## 用途

提高生词发现的准确度，比如在[单词发现者](https://chrome.google.com/webstore/detail/word-discoverer-expand-yo/noncaeikjgpbdeoocblijjgegnobogib?hl=zh-CN)中，生词识别和基于白名单的去除不太准确，如果支持 Lemmatization，就会好很多。

## 参考资料

- [词干提取 – Stemming | 词形还原 – Lemmatisation](https://easyai.tech/ai-definition/stemming-lemmatisation/)
- [NLP入门（三）词形还原（Lemmatization）](https://segmentfault.com/a/1190000016891462)
- [词形还原工具对比](https://www.zmonster.me/2016/01/21/lemmatization-survey.html)

**Comment by yoyoLooming** [comment]
谢谢, helpful!
